flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/assigners/MySqlSnapshotSplitAssigner.java

```diff
@@ -75,13 +76,14 @@ public class MySqlSnapshotSplitAssigner implements MySqlSplitAssigner {
     private final int currentParallelism;
     private final List<TableId> remainingTables;
     private final boolean isRemainingTablesCheckpointed;
+    private final Object lock = new Object();
+    private final AtomicReference<Throwable> uncaughtSplitterException;
```

Comment:
Could we use `volatile Throwable `?

Reply:
Sure these two share the same semantic. I'll change it to `volatile` for simplicity.